bluesky/tests/test_new_examples.py

```diff
@@ -508,6 +509,19 @@ def plan(*args):
     assert next(p) == 1
 
 
+def test_repeat():
+    # Check if lists and callables both work
+    messages = [1, 2, 3]
+
+    def plan():
+        yield from messages
+
+    num = 3
+    expected = [Msg('checkpoint'), 1, 2, 3] * num
+    assert list(repeat(plan, num=num)) == expected
+    assert list(repeat(messages, num=num)) == expected
```

Comment:
This will fail with `assert list(repeat(plan(), num=num)) == expected`

Just need a bit of caching logic if handed a 'live' generator (rather than a Sequence or a function)

Reply:
Right -- I think the terminology for "'live' generator" is "generator instance" (`plan()`) as opposed to a generator function (`plan`). This seems like a job for `itertools.tee`.

Reply:
@ZLLentz Do you have bandwidth to look at this? I'm happy to throw a commit on top of this to push it over the finish line if you are otherwise occupied.

Reply:
@danielballan Sorry for dropping this so close to the end without saying something. I think it's clear that it's just an `itertools.tee` away from accepting all forms of plan input. My bandwidth is low through the end of January.

Reply:
K, no worries, I'll tack that on and merge.